Tests/test_bgzf.py

```diff
@@ -126,9 +124,13 @@ def check_by_line(self, old_file, new_file, old_gzip=False):
                 h = open(old_file, mode)
             old = h.read()
             if "b" in mode:
-                old = _as_bytes(old)
+                if isinstance(old, str):
+                    import codecs
```

Comment:
Given we need it four times, I'd put the ``import codecs`` at the top of the file.